lib/chef_handler_foreman/foreman_facts.rb

```diff
@@ -55,7 +62,15 @@ def normalize(os)
     def plain_attributes
       # chef 10 attributes can be access by to_hash directly, chef 11 uses attributes method
       attributes = node.respond_to?(:attributes) ? node.attributes : node.to_hash
-      plainify(attributes.to_hash).flatten.inject(&:merge)
+      if @whitelist
+        attrs = plainify((attributes.select {|key| @whitelist.include?(key)}).to_hash).flatten.inject(&:merge)
```

Comment:
a bit more DRY version

```ruby
attributes = attributes.select { |key, value| @whitelist.include?(key) } if @whitelist
attrs = plainify((attributes.to_hash).flatten.inject(&:merge)
```

also note that two arguments in select block (key, value) are required to be compatible with ruby 1.8